### PR TITLE
Returns a thumbnail even when avatar is blacklisted

### PR DESCRIPTION
**Pull request by arcadianCdr**
Reading the doc, I expected I could combine "useAvatarAsThumbnail" and blacklist the avatar (to get only the thumbnail).

Since the two are handled by the same function and the check on avatar blacklist was at the very top, it did not work (I was getting either nothing or both avatar/thumbnail).

I made a small modification to the blacklist check to allow getting only a thumnail.

Note: I did not add support to blacklist thumbnails as it seemed redundant with the useAvatarAsThumbnail.

### CODE REVIEW

**Comment by arcadianCdr** [comment]
Ready for review again, with the test case and zodiac computation included.

File: plugins/freeones/main.ts [hunk 1]
```diff
@@ -212,13 +212,47 @@ module.exports = async (ctx: MyContext): Promise<ActorOutput> => {
...
+    const bornOn = $moment(getAge().bornOn);
+    if (!bornOn) {
+      $logger.verbose("No birth date found: zodiac will be empty");
+      return {};
+    }
+
+    const day = bornOn.date();
+    const month = bornOn.month() + 1;
+    let zodiacResult;
+    if ((month == 1 && day <= 20) || (month == 12 && day >= 22)) zodiacResult = signs.capricorn;
```
Comment (boi123212321) [inline plugins/freeones/main.ts:241]: Pretty sure this can all be refactored using some array logic

Comment (leadwolf) [reply plugins/freeones/main.ts:241]: and the core logic can maybe be moved to the top of the file, or somewhere else

File: plugins/freeones/main.ts [hunk 2]
```diff
@@ -276,8 +309,10 @@ module.exports = async (ctx: MyContext): Promise<ActorOutput> => {
     const imgId = await $createImage(url, `${actorName} (avatar)`);
 
     if (!useAvatarAsThumbnail) {
+      if (isBlacklisted("avatar")) return {};
       return { avatar: imgId };
     } else {
+      if (isBlacklisted("avatar")) return { thumbnail: imgId };
       return {
         avatar: imgId,
         thumbnail: imgId,
```
Comment (leadwolf) [inline plugins/freeones/main.ts:318]: I feel like all these ifs can be simplified or be more straightforward
```typescript
    if (isBlacklisted("avatar") && !useAvatarAsThumbnail) {
      // If not using either avatar or thumbnail, return nothing
      return {};
    }

    const imgId = await $createImage(url, `${actorName} (avatar)`);
    const result: Partial<{ avatar: string; thumbnail: string }> = {};

    if (!isBlacklisted("avatar")) {
      result.avatar = imgId;
    }
    if (useAvatarAsThumbnail) {
      result.thumbnail = imgId;
    }

    return result;
```

**Comment by arcadianCdr** [comment]
Voila. New version using array logic and simplified test structure proposed by leadwolf.

File: plugins/freeones/main.ts [hunk 3]
```diff
@@ -211,49 +211,45 @@ module.exports = async (ctx: MyContext): Promise<ActorOutput> => {
...
+      "Gemini",
+      "Cancer",
+      "Leo",
+      "Virgo",
+      "Libra",
+      "Scorpio",
+      "Sagittarius",
+    ];
+
+    return signAtMonthStart[day <= signSwitchDay[month] ? month : (month + 1) % 12];
```
Comment (leadwolf) [inline plugins/freeones/main.ts:236]: It's clever, but I didn't understand on the first read. Maybe extract that index to a descriptive variable name. And a comment explaining how the days and months relate. Ex:
signSwitchDay[i] gives the boundary day between two signs, for the month at index 'i'.
signSwitchDay[0] gives 20, meaning any day less than 20 at month index 0 (january) is the sign at the start of that month index (capricorn), and any day greater than 20 is the sign at the start of the next month: i+1 (aquarius)

**Review by leadwolf — APPROVED** [review]

File: plugins/freeones/main.ts (hunk 3 above)
Comment (arcadianCdr) [reply plugins/freeones/main.ts]: Hey, you guys wanted the array logic ;-) I've added your comments!

**Comment by arcadianCdr** [comment]
Now splitting hairs (and just for the sake of discussion, not for actual change): the zodiac calculation should normally take into account the place of birth and the precise full birth date, up to hour and minutes. So basically the whole concept of a fixed range valid in all cases like here is an approximation. Not all sources even agree on the same cutout dates, so in the end I modified the switch days to take Encyclopedia Britannica's ranges.

Personally, as a Cartesian logic driven dev, I could not care less about zodiac signs, and I'm perfectly happy with the "approximation" ;-)

**Review by leadwolf — APPROVED** [review]

**Pull request merged**